src/graphql/mutation/user.ts

```diff
@@ -17,3 +19,39 @@ export const createUser = async (_root, args) => {
   const savedUser = await newUser.save();
   return getAuthToken(savedUser);
 };
+
+export const loginWithGoogle = async (_root, args) => {
+  const client = auth.fromAPIKey(process.env.GOOGLE_SECRET!);
+  const { idToken } = args;
+  const res = await client.verifyIdToken({ idToken });
+  const { email } = <TokenPayload>res.getPayload();
+  return await useEmailToGetAuthToken(email!);
+};
+
+export const loginWithGithub = async (_root, args) => {
+  const { code } = args;
+  const credentials = {
+    code,
+    client_secret: process.env.GITHUB_SECRET,
+    client_id: process.env.GITHUB_CLIENT_ID,
+  };
+  let res = await requestGithubToken(credentials);
+  const { access_token } = res;
+  res = await requestGithubUserAccount(access_token);
+  const { email } = res;
+  return await useEmailToGetAuthToken(email);
+};
+
+const useEmailToGetAuthToken = async (email: string) => {
+  const user = await User.findOne({ email });
+  if (!user) {
+    const username = email.split("@")[0];
```

Comment:
The username is taken to be what comes before the @ symbol. Is that Okey or do you see any better approach? @Anu-Asok